src/goto-instrument/cover_instrument_other.cpp

```diff
@@ -75,10 +75,19 @@ void cover_instrument_end_of_function(
   const irep_idt &function_id,
   goto_programt &goto_program)
 {
-  auto if_it = goto_program.instructions.end();
-  while(!if_it->is_function_call())
-    if_it--;
-  if_it++;
+  const auto last_function_call = std::find_if(
+    goto_program.instructions.rbegin(),
+    goto_program.instructions.rend(),
+    [](const goto_programt::instructiont &instruction) {
+      return instruction.is_function_call();
+    });
+  INVARIANT(
+    last_function_call != goto_program.instructions.rend(),
+    "Goto program should have at least one function call");
```

Comment:
It looks like the reason that there is always a function call is that this function is only called for the entry point, which always calls the main function. We should probably say this in the invariant message.

Also nit: invariant messages should start with a lowercase letter.